source/MaterialXGenShader/OCIOColorManagementSystem.cpp

```diff
@@ -254,18 +437,61 @@ ImplementationPtr OCIOColorManagementSystem::getImplementation(const ColorSpaceT
         }
         */
 
+        // Cache binding information
+        _ocioInfo->updateUniformResources(shaderDesc);
+        _ocioInfo->updateTextureResources(shaderDesc);
+       
         // Create an implementation based on source code
         ImplementationPtr impl = _document->getImplementation(transformFunctionName);
         if (!impl)
         {
             impl = _document->addImplementation(transformFunctionName);
 
             // Note: There is only one input so we just use the default name: "inPixel".
-            impl->addInput("inPixel", typeName);
+            InputPtr pixelInput = impl->addInput(IN_PIXEL_STRING, typeName);
+            pixelInput->setIsUniform(false);
             impl->addOutput(outputName, typeName);
             impl->setAttribute("sourcecode", fullFunction);
             impl->setAttribute("function", functionName);
             impl->setAttribute("target", _ocioInfo->target);
+
+            // Add all of the additional uniforms as additional inputs on the impl so it can be accessed later one in the shader node ?
```

Comment:
Cache uniform information on the implementation for usage later on to create a CM uniform block.